R/RLum.Analysis-class.R

```diff
@@ -912,40 +915,78 @@ setMethod(
         message("[sort_RLum()]: Only the first field will be used in sorting")
         slot <- slot[1]
       }
-      sort.by.slot <- TRUE
     }
+
     if (!is.null(info_element)) {
       .validate_class(info_element, "character", extra = "NULL")
-      valid.names <- names(object@records[[1]]@info)
+      valid.names <- c("XY_LENGHT", "X_MIN", "X_MAX", "Y_MIN", "Y_MAX", names(object@records[[1]]@info))
       if (any(!info_element %in% valid.names)) {
         .throw_error("Invalid 'info_element' name, valid names are: ",
                      .collapse(valid.names))
       }
-      if (length(info_element) > 1) {
-        message("[sort_RLum()]: Only the first field will be used in sorting")
-        info_element <- info_element[1]
-      }
     }
-    if (is.null(slot) && is.null(info_element)) {
+
+    if (is.null(slot) && all(is.null(info_element))) {
       .throw_error("At least one of 'slot' and 'info_element' should not be NULL")
     }
-    .validate_logical_scalar(decreasing)
 
-    ## extract the values from the records according to which the sorting is
-    ## done: ordering by slot always supersedes ordering by info_element
-    vals <- if (sort.by.slot) {
-              sapply(object@records, function(x) slot(x, slot))
-            } else {
-              sapply(object@records, function(x) slot(x, "info")[[info_element]])
-            }
+    .validate_class(descending, classes = c("logical"))
+
+    ## recycle decreasing to match selection
+    descending <- rep(descending, length.out = sum(c(1, length(info_element))))
```

Comment:
So this is `length.out = length(info_element) + 1`?

Reply:
🤣 Yes, in essence.